dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Data.Sqlite;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel.Connectors.Sqlite;
+using Microsoft.SemanticKernel.Data;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Extension methods to register SQLite <see cref="IVectorStore"/> instances on an <see cref="IServiceCollection"/>.
+/// </summary>
+public static class SqliteServiceCollectionExtensions
+{
+    /// <summary>
+    /// Register a SQLite <see cref="IVectorStore"/> with the specified service ID
+    /// and where the SQLite <see cref="SqliteConnection"/> is retrieved from the dependency injection container.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="IVectorStore"/> on.</param>
+    /// <param name="options">Optional options to further configure the <see cref="IVectorStore"/>.</param>
+    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    /// <returns>Service collection.</returns>
+    public static IServiceCollection AddSqliteVectorStore(
+        this IServiceCollection services,
+        SqliteVectorStoreOptions? options = default,
+        string? serviceId = default)
+    {
+        services.AddKeyedTransient<IVectorStore>(
+            serviceId,
+            (sp, obj) =>
+            {
+                var connection = sp.GetRequiredService<SqliteConnection>();
+                var selectedOptions = options ?? sp.GetService<SqliteVectorStoreOptions>();
+
+                return new SqliteVectorStore(connection, options);
+            });
+
+        return services;
+    }
+
+    /// <summary>
+    /// Register a SQLite <see cref="IVectorStore"/> with the specified service ID.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="IVectorStore"/> on.</param>
+    /// <param name="connection"><see cref="SqliteConnection"/> that will be used to manage the data in SQLite.</param>
+    /// <param name="options">Optional options to further configure the <see cref="IVectorStore"/>.</param>
+    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    /// <returns>Service collection.</returns>
+    public static IServiceCollection AddSqliteVectorStore(
+        this IServiceCollection services,
+        SqliteConnection connection,
```

Comment:
The di registrations for the other connectors are typically taking either no client or taking the info to construct the client, e.g. connection string as parameters here.  Are we diverging here for a specific reason?

Reply:
In this implementation, it's not enough to initialize `SqliteConnection`. When it's initialized, vector search extension should be loaded, since it's not supported in SQLite out-of-the-box. There are a couple of ways to load the extension:
1. Using `SqliteConnection.LoadExtension` method - this method accepts a string parameter with absolute/relative file path to pre-compiled vector search extension (`.dll`) - we can do that on our side, as soon as we initialize `SqliteConnection`, we can call `LoadExtension` with provided extension file path from user.
2. Extension method can be loaded on SQLite side directly using `SELECT load_extension('extension_name');` command. 

Taking into account multiple ways of extension loading, I wasn't sure if we should perform it on our side or delegate it to the user and then just accept `SqliteConnection` ready for usage. Would be interesting to hear your opinion about that, thanks!